### PR TITLE
create a way to add some additional params to redirect url

### PR DESCRIPTION
**Pull request by pks1989**
Hi! I very like you library for lightweight. But custom auth server in my company wants to have some specific user info form redirect URL. Like deviceId, analytics systems id and other. And i don't see possibility do that with this lib.

### CODE REVIEW

File: src/client/authorization-code.ts [hunk 1]
```diff
@@ -56,7 +64,7 @@ export class OAuth2AuthorizationCodeClient {
    * Returns the URi that the user should open in a browser to initiate the
    * authorization_code flow.
    */
-  async getAuthorizeUri(params: GetAuthorizeUrlParams): Promise<string> {
+  async getAuthorizeUri(params: GetAuthorizeUrlParams, customParams?: GetAuthorizeUrlCustomParams): Promise<string> {
```
Comment (evert) [inline src/client/authorization-code.ts:67]: The need for this makes sense. `clientCredentials` also has some support for this, but there they are called `extraParams`. Could you take a look here:

https://github.com/badgateway/oauth2-client/blob/main/src/client.ts#L135

And make the naming convention and type structure identical to what is done here? Also, there's some security measures there that are important here too. We don't want users to accidentally overwrite parameters.

Here's the pull request that introduced that feature:

https://github.com/badgateway/oauth2-client/pull/91

**Review by evert — CHANGES_REQUESTED** [review]
Unittests would also be very welcome for this. It will make it easier for me to quickly merge and publish.

File: src/client/authorization-code.ts (hunk 1 above)
Comment (pks1989) [reply src/client/authorization-code.ts]: @evert Hi. Thanks you for review. I change naming and type structure. Аdd "core params" rewrite checking. 
Also add unit test for positive and negative cases of uses extraParams.

File: src/client/authorization-code.ts [hunk 2]
```diff
@@ -26,6 +26,11 @@ type GetAuthorizeUrlParams = {
    * List of scopes.
    */
   scope?: string[];
+
+  /**
+   * List of non-standard parameters that may need, if your auth server want to have some additional info in url
```
Comment (evert) [inline src/client/authorization-code.ts:31]:
```suggestion
   * Any parameters listed here will be added to the query string for the authorization server endpoint.
```

**Comment by evert** [comment]
Looks good! just some lint errors now

**Comment by pks1989** [comment]
> Looks good! just some lint errors now

Fixed :) All green now

**Comment by motnyk** [comment]
👋 @evert, just checking if you have any plans to release a 2.2.4 version of the client with this change anytime soon?

**Comment by evert** [comment]
Lets do it!

**Comment by evert** [comment]
Done
